Docs/ProjectSpec.md

```diff
@@ -164,6 +171,8 @@ This will provide default build settings for a certain product type. It can be a
 - app-extension.messages-sticker-pack
 - xpc-service
 
+Note: Passing "" for the product type implies you want a LegacyTarget
```

Comment:
What about if legacy targets need "legacy" as the type? Or alternatively that a "legacy" or "buildTool" dictionary must be defined? Thoughts?

Reply:
I like the idea of legacy as the type here. Feels more explicit than the presence of a build tool dictionary

Reply:
Unfortunately the product type is defined in `xcproj` not here in `XcodeGen` so we can't add a new variant easily (without building another enum on top of the existing one).